apps/api/scripts/calculate-next-season-eligibility.ts

```diff
@@ -235,37 +245,99 @@ Examples:
       `   (${totalForfeited.toString()} forfeited - ${totalSubsequentClaims.toString()} already claimed)\n`,
     );
 
-    // Step 6: Calculate individual rewards
+    // Step 6: Query for all agent boosters during the season.
+    const [season] = await db
+      .select()
+      .from(seasons)
+      .where(eq(seasons.number, seasonNumber));
+    if (!season) {
+      console.error(`Season ${seasonNumber} not found`);
+      process.exit(1);
+    }
+    const userAgentBoosts = await db
+      .select({
+        address: boostChanges.wallet,
+        boostAmount: sum(boostChanges.deltaAmount),
+      })
+      .from(agentBoosts)
+      .innerJoin(boostChanges, eq(agentBoosts.changeId, boostChanges.id))
+      .where(gte(agentBoosts.createdAt, season.startDate))
+      .groupBy(boostChanges.wallet);
+
+    const userAgentBoostsMap = userAgentBoosts.reduce(
+      (acc, boost) =>
+        acc.set(
+          BlockchainAddressAsU8A.decode(boost.address),
+          BigInt(boost.boostAmount || "0"),
+        ),
+      new Map<string, bigint>(),
+    );
+
+    // Step 7: Query for all users that had an agent compete in a competition during the season.
+    const userCompetitions = await db
+      .select({
+        address: users.walletAddress,
+        competitionIds: sql<
+          string[]
+        >`array_agg(DISTINCT ${competitions.id})`.as("competitionIds"),
+      })
+      .from(users)
+      .innerJoin(agents, eq(users.id, agents.ownerId))
+      .innerJoin(competitionAgents, eq(agents.id, competitionAgents.agentId))
+      .innerJoin(
+        competitions,
+        eq(competitionAgents.competitionId, competitions.id),
+      )
+      .where(
+        and(
+          gte(competitions.startDate, season.startDate),
+          lte(competitions.startDate, referenceTime),
+          eq(competitionAgents.status, "active"),
```

Comment:
Competition must have started after the beginning of the season and before the reference date. The user's agent must not be disqualified or withdrawn.